### PR TITLE
Add getters for delegators in client v2

### PR DESCRIPTION
**Pull request by limemloh**
## Purpose

Depends on https://github.com/Concordium/concordium-grpc-api/pull/17
Related to https://github.com/Concordium/concordium-node/issues/433

## Changes

- Support `GetPoolDelegators`, `GetPoolDelegatorsRewardPeriod`, `GetPassiveDelegators`, `GetPassiveDelegatorsRewardPeriod` and `GetBranches` in gRPC API v2.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [ ] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: examples/v2_get_branches.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+use structopt::StructOpt;
+
+#[derive(StructOpt)]
+struct App {
+    #[structopt(
+        long = "node",
+        help = "GRPC interface of the node.",
+        default_value = "http://localhost:10001"
+    )]
+    endpoint: tonic::transport::Endpoint,
```
Comment (abizjak) [inline examples/v2_get_branches.rs:14]: Please use `endpoints::Endpoint` instead in these examples, where `endpoints` is imported from concordium-rust-sdk.

THis makes the example usable if somebody just imports concordium-rust-sdk, and not tonic directly.

File: examples/v2_get_passive_delegators.rs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+use structopt::StructOpt;
+
+#[derive(StructOpt)]
+struct App {
+    #[structopt(
+        long = "node",
+        help = "GRPC interface of the node.",
+        default_value = "http://localhost:10001"
+    )]
+    endpoint: tonic::transport::Endpoint,
```
Comment (abizjak) [inline examples/v2_get_passive_delegators.rs:15]: Likewise.

File: examples/v2_get_passive_delegators_reward_period.rs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+use structopt::StructOpt;
+
+#[derive(StructOpt)]
+struct App {
+    #[structopt(
+        long = "node",
+        help = "GRPC interface of the node.",
+        default_value = "http://localhost:10001"
+    )]
+    endpoint: tonic::transport::Endpoint,
```
Comment (abizjak) [inline examples/v2_get_passive_delegators_reward_period.rs:15]: And here.

File: examples/v2_get_pool_delegators.rs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+use structopt::StructOpt;
+
+#[derive(StructOpt)]
+struct App {
+    #[structopt(
+        long = "node",
+        help = "GRPC interface of the node.",
+        default_value = "http://localhost:10001"
+    )]
+    endpoint: tonic::transport::Endpoint,
```
Comment (abizjak) [inline examples/v2_get_pool_delegators.rs:15]: And here.

File: src/types/mod.rs [hunk 5]
```diff
@@ -239,6 +239,26 @@ impl StakePendingChange {
     }
 }
 
+#[derive(Debug, Clone, Copy)]
+/// Stream item for `GetPoolDelegators`.
```
Comment (abizjak) [inline src/types/mod.rs:243]: This is not a useful comment. Please document the meaning of it.

File: src/types/mod.rs [hunk 6]
```diff
@@ -239,6 +239,26 @@ impl StakePendingChange {
...
+    /// The delegator account address.
+    pub account:        AccountAddress,
+    /// The amount of stake currently staked to the pool.
+    pub stake:          Amount,
+    /// Pending change to the current stake of the delegator.
+    pub pending_change: Option<StakePendingChange>,
+}
+
+#[derive(Debug, Clone, Copy)]
+/// Stream item for `GetPoolDelegatorsRewardPeriod`.
```
Comment (abizjak) [inline src/types/mod.rs:254]: Also here.

File: src/v2/generated.rs [hunk 7]
```diff
@@ -2176,3 +2202,18 @@ impl TryFrom<TokenomicsInfo> for super::types::RewardsOverview {
...
+    type Error = tonic::Status;
+
+    fn try_from(value: Branch) -> Result<Self, Self::Error> {
+        Ok(Self {
+            block_hash: value.block_hash.require()?.try_into()?,
+            children:   value
+                .children
+                .into_iter()
+                .map(|c| c.try_into())
+                .collect::<Result<_, _>>()?,
```
Comment (abizjak) [inline src/v2/generated.rs:2216]: I don't think this is a good instance. It is recursive, with no constraint on the depth of recursion. Please revise.

**Review by Bargsteen — COMMENTED** [review]
Looks fine to me. But the recursive TryFrom would be good to have fixed as Ales noted.

File: src/v2/generated.rs (hunk 7 above)
Comment (limemloh) [reply src/v2/generated.rs]: I have rewritten the try_from to avoid recursion and added a unit test

**Review by abizjak — APPROVED** [review]
I'm not super keen on the double traversal, but it requires some unsafe code to get rid of it I think. So let's leave it at t his.

**Review by Bargsteen — APPROVED** [review]

**Pull request merged**